language/move-prover/bytecode-to-boogie/src/code_writer.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A helper to emit code. Supports indentation and maintains source to target location information.
+
+use std::collections::{BTreeMap, Bound};
+
+use codespan::{ByteIndex, ColumnIndex, FileName, LineIndex};
+
+use ir_to_bytecode_syntax::ast::Loc;
```

Comment:
How specific to Move IR is this? It would be excellent to be able to reuse it for the source language if we can.

Reply:
Not at all. It does depend on type Loc = Span<ByteIndex>. We should be able to move if needed.